apps/admin/src/app/(api)/api/v1/mark_peaks/route.ts

```diff
@@ -0,0 +1,38 @@
+import { env } from "@/env.mjs";
+import { findAndMarkPeaks, getAllSensors } from "@energyleaf/db/query";
+import { type NextRequest, NextResponse } from "next/server";
+
+export const POST = async (req: NextRequest) => {
+    const reportApiKey = env.CRON_SECRET;
+    if (!req.headers.has("Authorization") || req.headers.get("Authorization") !== reportApiKey) {
+        return NextResponse.json({ status: 401, statusMessage: "Unauthorized" });
+    }
+
+    const startDate = new Date();
+    startDate.setHours(startDate.getHours() - 2, 0, 0, 0);
```

Comment:
Ist 2h nicht etwas krass weit hinten, um auf Peaks zu prüfen? Hätte maximal 30 min gedacht, eher so 10 oder so...

Kommt der bei setHours gegen 1 Uhr Nachts mit Minuswerten klar, also dass der 23 Uhr vom Vortag nimmt?

Reply:
Funktion läuft z.B. um 10 Uhr. Dann wird 8-9 Uhr verarbeitet. Wir können das auch auf 9-10 Uhr setzen.

Reply:
Ich denke alle 30 Minuten wäre besser und wäre weiter sogar für den Zyklus von alle 10 Minuten, damit die Daten schneller in die Oberfläche kommen. Zum Beispiel für den Use Case, dass man ne Anomalie-Mail bekommt und daraufhin die Oberfläche aufruft - das kann ja schnell bei unter 30 Minuten sein.

Ich meine in der Theorie könnte man das auch über einen Fenster-Ansatz ongoing im Imput Stream der Sensordaten lösen, aber ist denke ich overkill und passt so statisch alle x Min.